### ISSUE TITLE
What to do about Parsing?

### ISSUE THREAD

**Issue by kengdoj** [atbcb/ICTTestingBaseline#398]
It looks like [4.1 Parsing will be removed from WCAG 2.2](https://www.w3.org/WAI/WCAG22/Understanding/parsing). Not sure if WCAG 2.0 will receive a similar update.

Regardless, [Section 508 cites WCAG 2.0 date December 11, 2008](https://www.access-board.gov/ict/#702.10.1) in the Referenced Standards. An update to Section 508 is not expected any time soon. 

Do we keep [Baseline Test 24. Parsing](https://ictbaseline.access-board.gov/24Parsing/)?

**Comment by michaelhortongsa** [comment]
BLUF: No.

While we may need to keep it (due to date of standard) , can we not score
against it?

I think this would signal to agencies and vendors that we want to be as
forward leaning as possible.

Similarly, we should include 2.0 AAA, 2.1, and 2.2 in tools to enable
agencies to optionally require higher standards; ART, OpenACR, Baseline,
etc.

Thank you, and have an accessible day!

~Michael

U.S. General Services Administration

Michael Horton

Senior ICT Accessibility Specialist

Government-wide IT Accessibility Program

Office of Government-wide Policy

Mobile: 202-856-5226

*Buy. Build. Be Accessible.* | Section508.gov <https://www.Section508.gov>

**Comment by kengdoj** [comment]
From [WCAG 2 FAQ](https://www.w3.org/WAI/standards-guidelines/wcag/faq/#parsing411):

> Why is success criteria 4.1.1 Parsing obsolete in WCAG 2.2? What about Parsing in WCAG 2.0 and 2.1?
> Success criteria (SC) 4.1.1 Parsing is obsolete and removed from WCAG 2.2. Parsing was included in WCAG 2.0 to ensure that browsers and assistive technologies could accurately parse markup and content. Since then, specifications (such as HTML) and browsers have improved how they handle parsing errors. Also, previously assistive technology did their own markup parsing. Now they rely on the browser.
> 
> With today’s technology, accessibility issues that would have failed 4.1.1, will fail other criteria, such as Info and Relationships ([SC 1.3.1](https://w3c.github.io/wcag/understanding/info-and-relationships)) or Name, Role, Value ([SC 4.1.2](https://w3c.github.io/wcag/understanding/name-role-value)). Therefore 4.1.1 is no longer needed for accessibility.
> 
> We are exploring options for handling 4.1.1 in WCAG 2.0 and 2.1. We may deprecate it in those versions or add a note referencing the update in WCAG 2.2. In some cases, web content that is required to meet WCAG 2.0 or 2.1 may need to meet 4.1.1.

**Comment by kengdoj** [comment]
- https://www.w3.org/WAI/WCAG20/errata/

> Success Criterion [4.1.1](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#ensure-compat-parses) was originally adopted to address problems that assistive technology had directly parsing HTML. Since this criterion was written, the HTML Standard has adopted specific requirements governing how user agents must handle incomplete tags, incorrect element nesting, duplicate attributes, and non-unique IDs.
> 
> Although the HTML Standard treats some of these cases as non-conforming for authors, it is considered to "allow these features" for the purposes of this Success Criterion because the specification requires that user agents support handling these cases consistently. In practice, this criterion no longer provides any benefit to people with disabilities in itself.
> 
> Issues such as missing roles due to inappropriately nested elements or incorrect states or names due to a duplicate ID are covered by different Success Criteria and should be reported under those criteria rather than as issues with 4.1.1.
> 
> This criterion should be considered as always satisfied for any content using HTML or XML.

- https://www.w3.org/WAI/WCAG21/Understanding/parsing

> Note
> This criterion has been removed from WCAG 2.2. In WCAG 2.1 and 2.0 this Success Criterion should be considered as always satisfied for any content using HTML or XML.
> 
> Note
> Since this criterion was written, the HTML Standard has adopted specific requirements governing how user agents must handle incomplete tags, incorrect element nesting, duplicate attributes, and non-unique IDs.
> 
> Although the HTML Standard treats some of these cases as non-conforming for authors, it is considered to "allow these features" for the purposes of this Success Criterion because the specification requires that user agents support handling these cases consistently. In practice, this criterion no longer provides any benefit to people with disabilities in itself.
> 
> Issues such as missing roles due to inappropriately nested elements or incorrect states or names due to a duplicate ID are covered by different Success Criteria and should be reported under those criteria rather than as issues with 4.1.1.
